scripts/update-frontend-canister.sh

```diff
@@ -11,10 +11,106 @@ help()
    echo "Options:"
    echo "  -d, --development-build    build canister using cargo"
    echo "  -r, --release-build        build canister using linux/amd64 docker image"
+   echo "  -c, --changelog            update CHANGELOG.md with latest release"
    echo "  -h, --help                 print this help message"
    echo
 }
 
+update_changelog()
+{
+    REPO_ROOT=$(git rev-parse --show-toplevel)
+    CHANGELOG_PATH="$REPO_ROOT/CHANGELOG.md"
+    CHANGELOG_PATH_BACKUP="$REPO_ROOT/CHANGELOG.md.bak"
+    UNRELEASED_LOC=$(grep -nE "# UNRELEASED" $CHANGELOG_PATH | head -n 1 | cut -f1 -d:)
+    DEPENDENCIES_LOC=$(grep -nE "## Dependencies" $CHANGELOG_PATH | head -n 1 | cut -f1 -d:)
+    FRONTEND_CANISTER_LOC=$(grep -nE "### Frontend canister" $CHANGELOG_PATH | head -n 1 | cut -f1 -d:)
+    MODULE_HASH_LOC=$(grep -nE "\- Module hash: [a-f0-9]{64}" $CHANGELOG_PATH | head -n 1 | cut -f1 -d:)
+    LATEST_RELEASE_LOC=$(grep -nE "# \d+\.\d+\.\d+" $CHANGELOG_PATH | head -n 1 | cut -f1 -d:)
+    LINE_ABOVE_LAST_RELEASE=$(($LATEST_RELEASE_LOC - 1))
+    NEW_WASM_CHECKSUM=$(shasum -a 256 "$REPO_ROOT/src/distributed/assetstorage.wasm.gz" | awk '{print $1}')
+
+    if ! command -v gh &> /dev/null
+    then
+        echo "gh could not be found (brew install gh && gh auth login)"
+        exit
+    fi
+    PR_NUMBER=$(gh pr view --json number --jq '.number')
+    if [ -z "$PR_NUMBER" ]; then
+        LINK_TO_PR="https://github.com/dfinity/sdk/pull/YOU-NEED-TO-OPEN-A-PR-FIRST"
+    else
+        LINK_TO_PR="https://github.com/dfinity/sdk/pull/$PR_NUMBER"
+    fi
+
+    if [ -z "$UNRELEASED_LOC" ] || [ "$UNRELEASED_LOC" -gt "$LATEST_RELEASE_LOC" ]; then
+        echo "No \"# Unreleased\" section found in changelog, or \"# Unreleased\" section is not at the top of the changelog"
+        exit 1
+    fi
+
+    cp $CHANGELOG_PATH $CHANGELOG_PATH_BACKUP
```

Comment:
This means .github/workflows/shellcheck.yml doesn't check this script
```suggestion
    cp "$CHANGELOG_PATH" "$CHANGELOG_PATH_BACKUP"
```